### PR TITLE
feat(button): add icons - INNO-1167

### PR DESCRIPTION
**Pull request by emeryro**
# PR description

Add possibility to have icons in buttons.
Also add EU version of buttons (same as EC)

### CODE REVIEW

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit 1d2f7ced4ee9034bca239ae2fa2abb0a4b2424af

https://deploy-preview-797--europa-component-library.netlify.com

File: src/systems/ec/implementations/react/packages/button/Button.jsx [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 import React from 'react';
 import PropTypes from 'prop-types';
+import Icon from '@ecl/ec-react-component-icon/Icon';
 
 const Button = ({
   variant,
   typeAttribute,
   label,
+  icon,
   extraClasses,
```
Comment (yhuard) [inline src/systems/ec/implementations/react/packages/button/Button.jsx]: Please rename `extraClasses` by `className`

File: src/systems/ec/implementations/react/packages/button/Button.jsx [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
   extraAttributes,
```
Comment (yhuard) [inline src/systems/ec/implementations/react/packages/button/Button.jsx]: And don't use `extraAttributes` when you already have `...props` :)

File: src/systems/ec/implementations/react/packages/button/Button.jsx [hunk 3]
```diff
@@ -24,6 +26,8 @@ const Button = ({
     >
       <span className="ecl-button__container">
         <span className="ecl-button__label">{label}</span>
+        {icon &&
+          icon.icon && <Icon {...icon} extraClasses="ecl-button__icon" />}
```
Comment (yhuard) [inline src/systems/ec/implementations/react/packages/button/Button.jsx]: is it necessary to check `icon.icon`?

Comment (emeryro) [reply src/systems/ec/implementations/react/packages/button/Button.jsx]: Yes, it is, as icon has some default value (for size), so object is net empty

**Comment by emeryro** [comment]
extraClasses has been renamed to className in Button and Icon (as it was used here).
extraAttributes has been removed.

Ready for a second review

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit e9275a84d0c8a6e081e32d30cfc9a485dd40f9e4

https://deploy-preview-797--europa-component-library.netlify.com

File: src/systems/ec/implementations/react/packages/button/Button.jsx [hunk 4]
```diff
@@ -32,18 +24,18 @@ const Button = ({
 
 Button.propTypes = {
   variant: PropTypes.string,
-  typeAttribute: PropTypes.string,
+  type: PropTypes.string,
   label: PropTypes.string,
-  extraClasses: PropTypes.string,
-  extraAttributes: PropTypes.string,
+  icon: Icon,
```
Comment (yhuard) [inline src/systems/ec/implementations/react/packages/button/Button.jsx]: This is not a valid propType. 

> Warning: Button: type specification of prop `icon` is invalid; the type checker function must return `null` or an `Error` but returned a object. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).

`PropTypes.shape(Icon.propTypes)` should to the trick

Comment (emeryro) [reply src/systems/ec/implementations/react/packages/button/Button.jsx]: done

**Pull request merged**